websites/G/Google Meet/presence.ts

```diff
@@ -1,25 +1,167 @@
-		largeImageKey:
-			"https://cdn.rcd.gg/PreMiD/websites/G/Google%20Meet/assets/logo.png",
-		startTimestamp: time,
+function capitalizeFirstLetter(string: string) {
+	if (string) {
+		return (
+			string.trim().charAt(0).toUpperCase() +
+			string.trim().slice(1).toLowerCase()
+		);
+	}
+}
+let oldURL: string,
+	pdCache = {
+		name: "",
+		details: "",
+		state: "",
+		meetTitle: "",
+		meetDetails: "",
+		meetState: "",
 	};
 
-	if (document.location.pathname.toLowerCase() === "/") {
+function replacePlaceholders(
+	original: string,
+	mTime: string,
+	mTitle: string,
+	mPeople: string
+) {
+	if (!original) return "";
+
+	let tempString = original;
+
+	if (mTime) tempString = tempString.replace("%meettime%", `| time: ${mTime}`);
+	else tempString = tempString.replace("%meettime%", "");
+
+	if (mTitle)
+		tempString = tempString.replace("%meettitle%", `| title: ${mTitle}`);
+	else tempString = tempString.replace("%meettitle%", "");
+
+	if (mPeople) {
+		tempString = tempString.replace(
+			"%meetpeople%",
+			`| amount of attendants: ${mPeople}`
+		);
+	} else tempString = tempString.replace("%meetpeople%", "");
+
+	if (!tempString) return "";
+	else if (tempString?.toLowerCase()?.includes("ready to join?")) return "";
+	else if (tempString.at(0) === "|")
+		tempString = capitalizeFirstLetter(tempString.slice(2));
+	else if (tempString.at(1) === "|")
+		tempString = capitalizeFirstLetter(tempString.slice(3));
+	else tempString = capitalizeFirstLetter(tempString);
+
+	return tempString;
+}
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+			largeImageKey:
+				"https://cdn.rcd.gg/PreMiD/websites/G/Google%20Meet/assets/logo.png",
+			startTimestamp: browsingTimestamp,
+		},
+		[
+			usePresenceName,
+			hideInviteCode,
+			presenceName,
+			presenceDetail,
+			presenceState,
+		] = await Promise.all([
+			presence.getSetting<boolean>("usePresenceName"),
+			presence.getSetting<boolean>("hideInviteCode"),
+			presence.getSetting<string>("meetTitle"),
+			presence.getSetting<string>("meetDetail"),
+			presence.getSetting<string>("meetState"),
+		]),
+		{ href, pathname } = document.location;
+
+	if (pathname.toLowerCase() === "/") {
 		presenceData.details = "Initial page";
 		presenceData.state = "Just waiting";
-	} else {
-		presenceData.smallImageKey = Assets.VideoCall;
-		presenceData.details = "In a meeting";
-		presenceData.state = `${
-			(document.querySelector(".wnPUne") ?? document.querySelector(".uGOf1d"))
-				.textContent
-		} users in the room`;
+		return presence.setActivity(presenceData);
+	}
+
+	let meetTitle =
+		document.querySelector('[id="ucc-8"]')?.textContent ??
+		document.querySelector('[jsname="r4nke"]')?.textContent;
+
+	const meetTime = document.querySelector('[jsname="W5i7Bf"]')?.textContent,
+		meetPeople = document.querySelector(".uGOf1d")?.textContent;
+
+	if (
+		(!meetTitle ||
+			meetTitle?.includes("Read to join") ||
+			meetTitle?.includes("left the meeting") ||
```

Comment:
Just wondering, if you browser language is in another language than english doesn't google meet show different text? So not sure if this includes will succeed in those languages.

Reply:
I forgot to check woops